File: backend/verify_smtp.js
const tls = require('tls');

const config = {
    host: "mail.spencergreenhotel.com",
    port: 465,
    user: "reservasi@spencergreenhotel.com",
    pass: "Sp3nc3r@B4tu"
};

console.log(`Connecting to ${config.host}:${config.port} via TLS...`);

const socket = tls.connect(config.port, config.host, { rejectUnauthorized: false }, () => {
    console.log('✅ Connected to server securely!');
    console.log('Waiting for greeting...');
});

socket.setEncoding('utf8');

let phase = 'greeting';

socket.on('data', (data) => {
    console.log(`< ${data.trim()}`);

    // Simple state machine for SMTP login
    if (phase === 'greeting' && data.includes('220')) {
        console.log('> EHLO localhost');
        socket.write('EHLO localhost\r\n');
        phase = 'ehlo';
    }
    else if (phase === 'ehlo' && data.includes('250 ')) {
        console.log('> AUTH LOGIN');
        socket.write('AUTH LOGIN\r\n');
        phase = 'auth_login';
    }
    else if (phase === 'auth_login' && data.includes('334')) {
        // Username in base64
        const userB64 = Buffer.from(config.user).toString('base64');
        console.log(`> ${userB64} (Username)`);
        socket.write(`${userB64}\r\n`);
        phase = 'auth_user';
    }
    else if (phase === 'auth_user' && data.includes('334')) {
        // Password in base64
        const passB64 = Buffer.from(config.pass).toString('base64');
        console.log(`> [PASSWORD_HIDDEN]`);
        socket.write(`${passB64}\r\n`);
        phase = 'auth_pass';
    }
    else if (phase === 'auth_pass') {
        if (data.includes('235')) {
            console.log('\n✅ Login SUCCESSFUL! Credentials are correct.');
        } else {
            console.log('\n❌ Login FAILED. Check credentials.');
        }
        socket.end();
    }
});

socket.on('error', (err) => {
    console.error(`❌ Connection Error: ${err.message}`);
});

socket.on('end', () => {
    console.log('Connection closed.');
});

File: backend/verify_smtp.py
import smtplib
import ssl
import sys

# Credentials provided by user
SMTP_HOST = "mail.spencergreenhotel.com"
SMTP_PORT = 465
SMTP_USER = "reservasi@spencergreenhotel.com"
SMTP_PASSWORD = "Sp3nc3r@B4tu"
SENDER_EMAIL = SMTP_USER

print(f"Testing SMTP Connection to {SMTP_HOST}:{SMTP_PORT}...")
print(f"User: {SMTP_USER}")

try:
    context = ssl.create_default_context()
    
    if SMTP_PORT == 465:
        print("Attempting SMTP_SSL connection...")
        with smtplib.SMTP_SSL(SMTP_HOST, SMTP_PORT, context=context) as server:
            print("Connected to server.")
            print(f"Logging in as {SMTP_USER}...")
            server.login(SMTP_USER, SMTP_PASSWORD)
            print("✅ Login SUCCESSFUL!")
            
            # Optional: Verify we can send a test email to self
            # uncomment to test actual sending
            # msg = f"Subject: SMTP Test\n\nThis is a test email from the verification script."
            # server.sendmail(SENDER_EMAIL, SENDER_EMAIL, msg)
            # print("Test email sent to self.")
            
    else:
        print("Attempting SMTP (STARTTLS) connection...")
        with smtplib.SMTP(SMTP_HOST, SMTP_PORT) as server:
            print("Connected to server.")
            server.ehlo()
            if server.has_extn('STARTTLS'):
                print("Starting TLS...")
                server.starttls(context=context)
                server.ehlo()
            print(f"Logging in as {SMTP_USER}...")
            server.login(SMTP_USER, SMTP_PASSWORD)
            print("✅ Login SUCCESSFUL!")

except smtplib.SMTPAuthenticationError as e:
    print(f"❌ Authentication FAILED: {e}")
    print("Check your username and password.")
except smtplib.SMTPConnectError as e:
    print(f"❌ Connection FAILED: {e}")
    print("Server might be unreachable or blocking the connection.")
except Exception as e:
    print(f"❌ An error occurred: {e}")
    import traceback
    traceback.print_exc()
